modules/effects/src/main/scala/dev/profunktor/redis4cats/redis.scala

```diff
@@ -1314,7 +1315,40 @@ private[redis4cats] class BaseRedis[F[_]: FutureLift: MonadThrow: Log, K, V](
       _.eval[output.Underlying](
         script,
         output.outputType,
-        // see comment in eval above
+        // see comment in eval above.
+        keys.toArray[Any].asInstanceOf[Array[K with Object]],
+        values: _*
+      ).futureLift.map(output.convert(_))
+    )
+
+  override def evalReadOnly(script: String, output: ScriptOutputType[V]): F[output.R] =
+    async
+      .flatMap(
+        _.evalReadOnly[output.Underlying](
+          script.getBytes(StandardCharsets.UTF_8),
```

Comment:
Temporal workaround until https://github.com/redis/lettuce/pull/2868 is merged and released.